dask/array/core.py

```diff
@@ -45,41 +45,36 @@
 from ..sharedict import ShareDict
 
 
-def getarray(a, b, lock=None):
-    """ Mimics getitem but includes call to np.asarray
-
-    >>> getarray([1, 2, 3, 4, 5], slice(1, 4))
-    array([2, 3, 4])
-    """
+def getter(a, b, asarray=True, lock=None):
     if isinstance(b, tuple) and any(x is None for x in b):
         b2 = tuple(x for x in b if x is not None)
         b3 = tuple(None if x is None else slice(None, None)
                    for x in b if not isinstance(x, (int, long)))
-        return getarray(a, b2, lock)[b3]
+        return getter(a, b2, asarray=asarray, lock=lock)[b3]
 
     if lock:
         lock.acquire()
     try:
         c = a[b]
-        if type(c) != np.ndarray:
+        if asarray:
             c = np.asarray(c)
```

Comment:
Working with it as a boolean makes the optimizations easier to write and reason about. Could be done with a callable, but it'd be a bit trickier. Perhaps hold off until a use case presents itself? I'm not sure when that would be useful.